From eaca9c65cd5d3b67759aa47029e5433fb03d91a5 Mon Sep 17 00:00:00 2001
From: Chunguang Li <chunguang.li_1@nxp.com>
Date: Tue, 11 Oct 2022 02:11:26 +0800
Subject: [PATCH] Add judgement for null return value.

Add the return value judgement when the return value
is Null of follow_up_info_extract then stop the flow
processing and return.

The issue detail info is as below:

CID 20810173 (#1 of 1): Dereference null return value (NULL_RETURNS)
5. dereference: Dereferencing fui, which is known to be NULL.

Signed-off-by: Chunguang Li <chunguang.li_1@nxp.com>
---
 port.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/port.c b/port.c
index 55dcf5a..65855c7 100644
--- a/port.c
+++ b/port.c
@@ -1526,6 +1526,8 @@ static void port_syfu_relay_info_insert(struct port *p,
 
 	if (syfu_relay->avail == 0)
 		return;
+	if (!fui)
+		return;
 
 	fup->follow_up.preciseOriginTimestamp =
 		tmv_to_Timestamp(syfu_relay->precise_origin_ts);
-- 
2.17.1

